File: 199A_Hexadecimals-Theorem/solution.cpp
/*
 * Sat 25 Mar 2023 02:04:58 AM UTC
 * https://codeforces.com/problemset/problem/199/A
*/

#include <bits/stdc++.h>

using namespace std;

#define OPTIMIZE(); ios::sync_with_stdio(0);cin.tie(0);
#define endl '\n'
#define int int64_t

signed main(){
    OPTIMIZE();

    int n;
    cin >> n;

    int a=0;
    int b=1;

    vector<int> nums;

    if(n==0){
        cout << "0 0 0";
        return 0;
    }

    if(n==1){
        cout << "0 1 0";
        return 0;
    }

    if(n==2){
        cout << "1 1 0";
        return 0;
    }

    while(n >= b){
        nums.push_back(a);
        int newfib = a+b;
        a = b;
        b=newfib;
    }

    if(nums.size() >= 3 && nums.back() + nums[nums.size()-3] + nums[nums.size()-4] == n){
        cout << nums.back() << " " << nums[nums.size()-3] << " " << nums[nums.size()-4];
        return 0;
    }

    cout << "I'm too stupid to solve this problem";


    return 0;
}

File: 199A_Hexadecimals-Theorem/solution.py
# Sat 25 Mar 2023 09:32:32 PM UTC
# https://codeforces.com/contest/199/problem/A

n = int(input())

a=0
b=1

fibs = [a,b]

while(b<n):
    newfib = a+b;
    a = b
    b = newfib
    fibs.append(b)

if(n <= 2):
  if(n == 0):
      print("0 0 0 ")
  elif(n == 1):
      print("0 0 1")
  elif(n == 2):
      print(" 0 1 1")
else:
    test = fibs[-1-1] + fibs[-1-3] + fibs[-1-4]

    if n==test:
      print("{} {} {}".format(fibs[-1-1], fibs[-1-3], fibs[-1-4]))
    else:
      print("I'm too stupid to solve this problem")



